include/zephyr/devicetree.h

```diff
@@ -4051,11 +4097,29 @@
  * @param ... variable number of arguments to pass to @p fn
  *
  * @see DT_INST_FOREACH_STATUS_OKAY
+ * @see DT_COMPAT_FOREACH_STATUS_OKAY_VARGS
  */
 #define DT_INST_FOREACH_STATUS_OKAY_VARGS(fn, ...) \
-	COND_CODE_1(DT_HAS_COMPAT_STATUS_OKAY(DT_DRV_COMPAT),	\
-		    (UTIL_CAT(DT_FOREACH_OKAY_INST_VARGS_,	\
-			      DT_DRV_COMPAT)(fn, __VA_ARGS__)),	\
+	COND_CODE_1(DT_HAS_COMPAT_STATUS_OKAY(DT_DRV_COMPAT),   \
+		    (UTIL_CAT(DT_FOREACH_OKAY_INST_VARGS_,      \
+			      DT_DRV_COMPAT)(fn, __VA_ARGS__)), \
+		    ())
+
+/**
+ * @brief Call @p fn on all nodes with compatible `compat`
+ *        and status `okay` with multiple arguments
+ *
+ *
+ * @param fn Macro to call for each enabled node. Must accept a
+ *           devicetree compatible and instance number as its only parameters.
+ * @param ... variable number of arguments to pass to @p fn
+ *
+ * @see DT_INST_FOREACH_STATUS_OKAY_VARGS
+ */
+#define DT_COMPAT_FOREACH_STATUS_OKAY_VARGS(compat, fn, ...)     \
+	COND_CODE_1(DT_HAS_COMPAT_STATUS_OKAY(compat),           \
+		    (UTIL_CAT(DT_FOREACH_OKAY_INST_VARGS_,       \
+			      compat)(fn, compat, __VA_ARGS__)), \
```

Comment:
Will recommend a review from DT maintainers for this newly added helper, I think this should have a corresponding test under `tests/lib/devicetree/api/` 
CC: @cfriedt @mbolivar-ampere @mbolivar

Reply:
the test was added

Reply:
Please move this out of the instance APIs (see the ` * @defgroup devicetree-inst Instance-based devicetree APIs` above -- this section of the API is for the DT_INST_FOO() macros)